WTG.Analyzers/Analyzers/BooleanLiteral/ArgumentListExtensions.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace WTG.Analyzers
+{
+	static class ArgumentListExtensions
+	{
+		public static IParameterSymbol? TryFindCorrespondingParameterSymbol(this ArgumentListSyntax argumentList, int index, SemanticModel semanticModel, CancellationToken cancellationToken)
+		{
+			var methodInvocation = argumentList.FirstAncestorOrSelf<InvocationExpressionSyntax>();
```

Comment:
Just look at the immediate parent, you could get misleading results if you walk too far...
```csharp
Foo(new Bar(true, false));
```

This also raises the question, "Should constructor arguments also be bound by this?"

Reply:
Wouldn't **First**AncestorOrSelf stop that? Or is a constructor not considered an invocation?

Reply:
ah, I see. I would say its at least useful for constructor arguments such as `leaveOpen:` in stream reader classes, not sure about others in general. For now I've fixed this crashing at least, but do you think we should also check constructor args?